presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -478,6 +492,66 @@ protected Node visitShowCreate(ShowCreate node, Void context)
             throw new UnsupportedOperationException("SHOW CREATE only supported for tables and views");
         }
 
+        @Override
+        protected Node visitShowCreateFunction(ShowCreateFunction node, Void context)
+        {
+            QualifiedFunctionName functionName = qualifyFunctionName(node.getName());
+            Collection<? extends SqlFunction> functions = metadata.getFunctionManager().getFunctions(session.getRequiredTransactionId(), functionName);
+            if (node.getParameterTypes().isPresent()) {
+                List<TypeSignature> parameterTypes = node.getParameterTypes().get().stream()
+                        .map(TypeSignature::parseTypeSignature)
+                        .collect(toImmutableList());
+                functions = functions.stream()
+                        .filter(function -> function.getSignature().getArgumentTypes().equals(parameterTypes))
+                        .collect(toImmutableList());
+            }
+            if (functions.isEmpty()) {
+                String types = node.getParameterTypes().isPresent()
+                        ? format("(%s)", Joiner.on(", ").join(node.getParameterTypes().get()))
+                        : "";
+                throw new PrestoException(FUNCTION_NOT_FOUND, format("Function not found: %s%s", functionName, types));
+            }
+
+            ImmutableList.Builder<Expression> rows = ImmutableList.builder();
+            for (SqlFunction function : functions) {
+                if (!(function instanceof SqlInvokedFunction)) {
+                    throw new PrestoException(GENERIC_USER_ERROR, "SHOW CREATE FUNCTION only supported for SQL functions");
+                }
+
+                SqlInvokedFunction sqlFunction = (SqlInvokedFunction) function;
+                CreateFunction createFunction = new CreateFunction(
+                        node.getName(),
+                        false,
+                        sqlFunction.getParameters().stream()
+                                .map(parameter -> new SqlParameterDeclaration(new Identifier(parameter.getName(), true), parameter.getType().toString()))
```

Comment:
Wow so we actually rebuild the create function? I thought we remember it in the DB at creation time and just return it.

Reply:
Yes, we're storing different component of a SQL function as separate columns in the DB, as opposed to the entire `CREATE FUNCTION` statement.